datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -3810,53 +3826,53 @@ mod tests {
 
     #[test]
     fn simplify_expr_case_when_first_true() {
-        // CASE WHEN true THEN 1 ELSE x END --> 1
+        // CASE WHEN true THEN 1 ELSE c1 END --> 1
         assert_eq!(
             simplify(Expr::Case(Case::new(
                 None,
                 vec![(Box::new(lit(true)), Box::new(lit(1)),)],
-                Some(Box::new(col("x"))),
+                Some(Box::new(col("c1"))),
             ))),
             lit(1)
         );
 
-        // CASE WHEN true THEN col("a") ELSE col("b") END --> col("a")
+        // CASE WHEN true THEN col('a') ELSE col('b') END --> col('a')
```

Comment:
I was worried about this change, as it seems to potentially change the intent of the test -- to use literals rather than column references. 

However, I see the issue is that the DataTypes need to match and after spending some time rewriting these tests to use column references rather than literals I think the literals are fine